crates/re_viewer_context/src/space_view/view_part_system.rs

```diff
@@ -12,26 +14,28 @@ use crate::{
 pub trait ViewPartSystem {
     // TODO(andreas): This should be able to list out the ContextSystems it needs.
 
-    /// The archetype queried by this scene element.
-    fn archetype(&self) -> ArchetypeDefinition;
+    /// Returns the minimal set of components that the system _requires_ in order to be instantiated.
+    fn required_components(&self) -> IntSet<ComponentName>;
```

Comment:
I think we need to clarify here wether or not this should include the archetype indicator component.

I'd argue it should _not_ include it, because most systems are compatible based on the set of actual component, and the indicator is just part of the heuristic of which system is assigned _by default_.

…but that tells me that the indicator component(s) (if any) should probably be communicated some how, separately, since it is such a common thing. For instance, the trait could look like this:


```rs
trait ViewPartSystem {
    /// We need these to show. Do not include indicator components.
    fn required_components(&self) -> ComponentNameSet;

    /// These are not required, but if these are found,
    /// it is a strong indication that this system should be active
    /// if also the required components are found.
    fn indicator_components(&self) -> ComponentNameSet {
        Default::default()
    }
 
    /// Called for all entities that have the required_components.
    /// Return `true` if this this is a good system to show by default for this entity.
    fn filter(&self, entity_components: &ComponentNameSet, store, …) -> bool {
        let indicator_components = self.indicator_component();
        if indicator_components.is_empty() {
            true
        } else {
            let has_any_indicator_component = 
                !entity_components.intersect(self.indicator_component()).is_empty();
            has_any_indicator_component
        }
    }
}
```

most systems would only need to implement the `required_components` and `indicator_components`